presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/StaticMetastoreLocator.java

```diff
@@ -138,34 +152,46 @@ private static URI checkMetastoreUri(URI uri)
         return uri;
     }
 
-    private static class Backoff
+    static class Backoff
     {
-        private static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
-        private static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
+        static final long MIN_BACKOFF = new Duration(50, MILLISECONDS).roundTo(NANOSECONDS);
+        static final long MAX_BACKOFF = new Duration(60, SECONDS).roundTo(NANOSECONDS);
 
-        private final Ticker ticker = Ticker.systemTicker();
+        private final Ticker ticker;
         private long backoffDuration = MIN_BACKOFF;
-        private OptionalLong lastFailureTimestamp = OptionalLong.empty();
+        private boolean success = true;
+        private long lastFailureTimestamp;
+
+        Backoff(Ticker ticker)
+        {
+            this.ticker = requireNonNull(ticker, "ticker is null");
+        }
 
         synchronized void fail()
         {
-            lastFailureTimestamp = OptionalLong.of(ticker.read());
+            success = false;
+            lastFailureTimestamp = ticker.read();
```

Comment:
This looks like incrementing `lastFailureTimestamp` value, but it may actually decrement it -- ticker can return a negative value.
Maybe init lastFailureTimestamp to Long.MIN_VALUE to make reader's life easier?

Reply:
`ticker can return a negative value`

Good point I did not know that. I will use MIN_VALUE